subprojects/language-native/src/main/java/org/gradle/language/assembler/tasks/Assemble.java

```diff
@@ -115,9 +116,10 @@ public void assemble() {
         setDidWork(result.getDidWork() || cleanedOutputs);
     }
 
-    @PathSensitive(PathSensitivity.RELATIVE)
     @InputFiles
     @SkipWhenEmpty
+    @IgnoreEmptyDirectories
+    @PathSensitive(PathSensitivity.RELATIVE)
     public ConfigurableFileCollection getSource() {
         return source;
     }
```

Comment:
How come `Assemble` was not created originally as a `SourceTask`? I think this is exactly the pattern we want to replace `SourceTask` with eventually.

Reply:
Same for `AbstractNativeCompileTask`... I mean, it's good that they are not `SourceTask`s, given our current plans, but why aren't they (given the past ideas about using task type hierarchies)?

Reply:
None of the native tasks are source tasks. I don't know why exactly that happened. Maybe because source task was already a pretty strange beast when we started implementing the native tasks.

Reply:
If I remember correctly, there were already suspicions on removing the `SourceTask`. I think Daz would have more insight on this.